src/adapters/philips_hue/mod.rs

```diff
@@ -123,9 +126,20 @@ struct HueLightService<T> {
 }
 
 impl<T: Controller> HueLightService<T> {
-    fn new(controller: T, id: u32, light: HueLight) -> Self {
+    fn new(mut controller: T, id: u32, light: HueLight) -> Self {
         debug!("Creating HueLightService {} for HueLight {:?}", id, light);
-        let service_id = Uuid::new_v4().to_simple_string();
+
+        // Get stored service ID if available, default to random UUID
```

Comment:
why do you need to store the service ID since you get it from light.get_unique_id() ?